src/parser/parser_support.c

```diff
@@ -11513,16 +11557,27 @@ pt_convert_dblink_dml_query (PARSER_CONTEXT * parser, PT_NODE * node, char *sql_
   int tmp_server_cnt = snl->server_cnt;
 
   PT_NODE *sub_sel = NULL;	/* for select sub-query */
-  PT_NODE *spec, *tbl_spec = NULL, *into_spec = NULL, *upd_spec = NULL, *server;
+  PT_NODE *list = NULL;		/* for insert select list */
+  PT_NODE *spec, *into_spec = NULL, *upd_spec = NULL, *server;
 
   switch (node->node_type)
     {
     case PT_INSERT:
       into_spec = node->info.insert.spec;
       sub_sel = node->info.insert.value_clauses;
-      if (sub_sel->info.node_list.list && sub_sel->info.node_list.list->node_type == PT_SELECT)
+      for (list = sub_sel->info.node_list.list; list; list = list->next)
 	{
-	  upd_spec = sub_sel->info.node_list.list->info.query.q.select.from;
+	  if (list->node_type == PT_SELECT)
```

Comment:
insert into x values (5 + (select count(*) from dual@srv1) );

If you perform the query as above, it seems that SELECT alone will not solve it.